pkg/generate/app/imagestreamlookup.go

```diff
@@ -109,7 +109,7 @@ func InputImageFromMatch(match *ComponentMatch) (*ImageRef, error) {
 		if err != nil {
 			return nil, err
 		}
-		input.AsImageStream = true
+		input.AsImageStream = !match.LocalOnly
```

Comment:
maybe never, i was having some trouble following the flow into this call so when i figured out the change i figured it was safer to do both of them.  but if you are confident it isn't a valid combination, i can undo this one.